apps/frontend/src/lib/components/LanguageSwitcher.svelte

```diff
@@ -0,0 +1,42 @@
+<script lang="ts">
+	import { locale, availableLocales } from '$lib/stores/locale';
+	import { setLocale, type Locale } from '$lib/paraglide/runtime.js';
+	import { Button, Dropdown, DropdownItem } from 'flowbite-svelte';
+	import { ChevronDownOutline } from 'flowbite-svelte-icons';
+
+	const localeNames: Record<string, string> = {
+		en: 'English',
+		nl: 'Dutch',
+		hi: 'हिन्दी'
+	};
+
+	let currentLocale = $derived($locale);
+
+	async function switchLanguage(newLocale: Locale) {
+		try {
+			setLocale(newLocale, { reload: true });
+		} catch (err) {
+			console.warn('Language switch failed, fallback reload:', err);
+			setLocale('en', { reload: true });
+		}
+	}
+</script>
+
+<Button>
+	{localeNames[currentLocale] || currentLocale}
+	<ChevronDownOutline class="ms-2 h-6 w-6 text-white dark:text-white" />
+</Button>
+<Dropdown>
+	{#each availableLocales as localeCode (localeCode)}
+		<DropdownItem onclick={() => switchLanguage(localeCode)}>
+			<span
+				class="flex w-full items-center justify-between text-sm font-medium text-gray-900 dark:text-white"
+			>
+				{localeNames[localeCode as Locale] || localeCode}
```

Comment:
What we _actually_ want here is to have translations of the language names themselves be part of the messages. 

We can call `m.locale_name({locale: localeCode})` (= name of language _in that language_). Together with iterating over pre-existing locales this would make locales/translations be fully self-configuring, that is, you can just add languages JSON files and voilá, another language supported.

Reply:
https://linear.app/younghumanclub/issue/PAI-1091/translate-names-of-languages